### ISSUE TITLE
Release 0.9

### ISSUE THREAD

**Issue by sk1p** [LiberTEM/LiberTEM#1183]
## TODO: Issues and pull requests to be fixed/finished before release

- [x] First, merge `DelayedJobExecutor` #1170 
- [x] Merge #1185, #1176 and related version bumps
- [x] TVIPS #1179 
- [x] Build a release candidate
- [x] #1196 
- [x] #1193 
- [x] #1197 
- [x] Docs say that only 1-bit and 6-bit work for MIB quad data. However, opening /Users/weber/LargeData/ER-C-1/groups/data_science/data/reference/MIB/quad/Au_focused/12-bit_raw-20210728_095715/12-bit_raw.hdr works, apparently. @sk1p could you perhaps comment on the status? :-)
* [x] Docs review: Add docstrings to properties `tiling_scheme_idx` and `sig_slice` of `UDFMeta`.
* [x] #1218 

@uellue @matbryan52 feel free to add additional tasks in this list

## Release checklist

* [x] Review open issues and pull requests
* [x] Run full CI pipeline, including slow tests, on [Azure DevOps](https://dev.azure.com/LiberTEM/LiberTEM/_build?definitionId=3)
* [ ] Handle deprecation, search the code base for `DeprecationWarning`
      that are supposed to be removed in that release.
* [ ] GUI dependency update with `npm install`
* [ ] Review https://github.com/LiberTEM/LiberTEM/security/dependabot and update dependencies
* [x] Full documentation review and update
   * [x] Link check using ``sphinx-build -b linkcheck "docs/source" "docs/build/html"`` #1197 
* [x] Run complete test suite, including slow tests that are deactivated by default
      and tests that require sample files.
* [x] Update the expected version in notes on changes, i.e. from `0.3.0.dev0`
      to `0.3.0` when releasing version 0.3.0. #1197
* [x] Update and review change log in `docs/source/changelog.rst`, merging
      snippets in `docs/source/changelog/*/` as appropriate. #1197 
* [x] Update the JSON files in the ``packaging/`` folder with author and project information
* [N/A] Edit `pytest.ini` to exclude flaky tests temporarily from release builds (note: we currently don't have tests marked as flaky)
* [x] Create a release candidate using `scripts/release`. See `scripts/release --help` for details.
* [x] `Confirm that wheel, tar.gz, and AppImage are built for the release candidate on
      GitHub <https://github.com/LiberTEM/LiberTEM/releases>`_
* [x] Confirm that a new version with the most recent release candidate is created in the
      `Zenodo.org sandbox <https://sandbox.zenodo.org/record/367108>`_ that is ready for submission.
* [x] Install release candidate packages in a clean environment
      (for example:
      `python -m pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple 'libertem==0.9.0rc1'`)
* [x] Test the release candidate docker image
* [x] For the GUI-related items, open in an incognito window to start from a clean slate
* [x] Correct version info displayed in info dialogue?
* [x] Link check in version info dialogue
* [x] Make sure you have test files of all supported types available
    * [] Include floats, ints, big endian, little endian, complex raw data
* [x] Open each test file
    * [x] Are parameters recognized correctly, as far as implemented?
    * [x] Any bad default values?
    * [x] Does the file open correctly?
    * [x] Have a look at the dataset info dialogue. Reasonable values?
* [x] Perform all analyses on each test file.
    * [x] Does the result change when the input parameters are changed?
    * [x] All display channels present and looking reasonable?
    * [x] Reasonable performance?
    * [x] Use pick mode.
* [x] Re-open all the files
    * [ ] Are the files listed in "recent files"?
    * [ ] Are the parameters filled from the cache correctly?
* [ ] Try opening all file types with wrong parameters
    * [ ] Proper understandable error messages?
* [x] Pick one file and confirm keyboard and mouse interaction for all analyses
    * [ ] Correct bounds check for keyboard and mouse?
* [ ] Check what happens when trying to open non-existent files or directories in the GUI.
    * [ ] Proper understandable error message?
    * [ ] Possible to continue working?
* [x] Shut down libertem-server while analysis is running
    * [x] Shut down within a few seconds?
    * [x] All workers reaped?
* [ ] Check what happens when trying to open non-existent files by scripting.
    * [ ] Proper understandable error message? TODO automate?
* [ ] Check what happens when opening all file types with bad parameters by scripting
    * [ ] Proper understandable error message? TODO automate?
* [x] Run libertem-server on Windows, connect to a remote dask cluster running on Linux,
  open all file types and perform an analysis for each file type.
* [x] Use the GUI while a long-running analysis is running
    * [x] Still usable, decent response times?
* [x] Confirm that pull requests and issues are handled as intended, i.e. milestoned and merged
  in appropriate branch.
* [x] Final version bump: `./scripts/release bump v0.9.0 --tag`, push to github
* [x] After pipeline finishes, write minimal release notes for the [release](https://github.com/liberTEM/LiberTEM/releases) and publish the GitHub release

## After releasing on GitHub

* [x] Confirm that all release packages are built and release notes are up-to-date
* [x] Install release package
* [x] Confirm correct version info
* [x] Confirm package upload to PyPI
* [x] Confirm images and tags on https://hub.docker.com/r/libertem/libertem
* [x] Publish new version on zenodo.org
* [x] Update documentation with new links, if necessary #1218 
    * [x] This release only: Update Azure badge: image https://dev.azure.com/LiberTEM/LiberTEM/_apis/build/status/LiberTEM.LiberTEM-data?branchName=master, Link https://dev.azure.com/LiberTEM/LiberTEM/_build?definitionId=4
    * [x] Add zenodo badge for the new release to Changelog page
* [x] Send announcement message on mailing list
* [N/A] Edit `pytest.ini` to include flaky tests again
* [x] Bump version in master branch to next .dev0 (`./scripts/release bump v0.X.0.dev0 --commit`)
* [x] Add to institutional publication databases
* [x] Add the current LiberTEM version to [CVL](https://github.com/Chasdfracterisation-Virtual-Laboratory/CharacterisationVL-Software>) - add both the singularity and the .desktop file!

**Comment by sk1p** [comment]
CI build w/ slow tests and recreating tox environments: https://dev.azure.com/LiberTEM/LiberTEM/_build/results?buildId=1648&view=results

**Comment by sk1p** [comment]
`tests/udf/test_memleak.py::test_executor_memleak[dask]` fails on Mac OS X for some reason, log here: https://dev.azure.com/LiberTEM/LiberTEM/_build/results?buildId=1648&view=logs&j=7f5f2969-2f30-5453-4239-97ef5e9a1fd4&t=abbc44e5-2855-5024-6cb4-fa218810c41b

Might need some tweaks to make the test more robust, or we really do have a memory leak?

**Comment by sk1p** [comment]
Another `-m slow` run here: https://dev.azure.com/LiberTEM/LiberTEM/_build/results?buildId=1655&view=results

**Comment by sk1p** [comment]
[v0.9.0rc1](https://github.com/LiberTEM/LiberTEM/releases/tag/v0.9.0rc1) published.

**Comment by sk1p** [comment]
Docker image runs fine with both docker itself and singularity, as documented.

**Comment by sk1p** [comment]
As does the AppImage.

**Comment by sk1p** [comment]
Looking at [the issues and PRs closed with the 0.9 milestone](https://github.com/LiberTEM/LiberTEM/milestone/14?closed=1), I think the changelog fragments are pretty much complete.

A minor issue missing from the changelog is #753, which was implemented as part of #1129

**Comment by uellue** [comment]
@sk1p @matbryan52 in https://github.com/LiberTEM/LiberTEM/pull/1197/commits/b0a5a0a230cbf84405985378a856b372837229ba I have added new contributors. Am I missing anyone?

**Comment by sk1p** [comment]
> Docs say that only 1-bit and 6-bit work for MIB quad data. However, opening /Users/weber/LargeData/ER-C-1/groups/data_science/data/reference/MIB/quad/Au_focused/12-bit_raw-20210728_095715/12-bit_raw.hdr works, apparently. 

Indeed, sorry about that - 12bit quad raw support was added a bit later than the others, and is indeed supported! Could you fix this in #1197? Good catch!

>  I have added new contributors. Am I missing anyone?

Looks good to me.

**Comment by uellue** [comment]
> Could you fix this in #1197?

Sure, will do! :-)

**Comment by uellue** [comment]
Discussion with @sk1p about the CI situation: So far there wasn't any substantial progress to get Azure Pipelines running again. In order to get the release out, @sk1p will migrate all Linux-based jobs to the data pipeline. We'll run Windows and OS X tests by hand once all pending PRs are merged, and continue with the release checklist.

For the time after, we'll deploy a dedicated machine to run all Linux CI jobs in-house. For OS X and Windows jobs we'd migrate to GitHub Actions. At the same time we'll wrap them in such a way that they are easy to execute locally with all Python versions. That way we have a fallback at hand. Should GitHub actions run into the same issues, we can think about actually deploying our own CI workers for that as well.

In conclusion, an expensive lession how depending on cloud services can lead to issues. I'm now more and more in favor to run open source solutions on own hardware with minimum exposure to the whims of third parties.

**Comment by sk1p** [comment]
I'll re-open this until the release is actually out there.

**Comment by uellue** [comment]
From my side the RC3 is good to go. :-)

**Comment by sk1p** [comment]
I'll go ahead and tag the release, then!

**Comment by sk1p** [comment]
https://github.com/Characterisation-Virtual-Laboratory/CharacterisationVL-Software/pull/136 update on CVL is in the pipeline

**Comment by uellue** [comment]
Congrats and thanks, everyone! Release 0.9 is officially finished. :-) Lots of work, nice result!
